packages/modal/src/Modal.js

```diff
@@ -0,0 +1,226 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { Transition } from 'react-transition-group'
+import { color, width, height } from 'styled-system'
+import { DialogOverlay, DialogContent } from '@reach/dialog'
+import { Box, CloseButton, Flex } from 'pcln-design-system'
+
+const Overlay = styled(DialogOverlay)`
+  background-color: rgba(0, 0, 0, 0.7);
+  bottom: 0;
+  left: 0;
+  overflow-y: auto;
+  position: fixed;
+  right: 0;
+  top: 0;
+  z-index: ${props => props.zindex || 100};
+  font-family: ${props => props.theme.font};
+  opacity: 0;
+  transition: opacity .5s cubic-bezier(0.50, 0.00, 0.25, 1.00);
+
+  ${props =>
+    props.transitionstate === 'entering' &&
+    `
+    opacity: 0;
+  `}
+
+  ${props =>
+    props.transitionstate === 'entered' &&
+    `
+    opacity: 1;
+  `}
+
+  ${props =>
+    props.transitionstate === 'exiting' &&
+    `
+    opacity: 0;
+  `}
+
+  ${props =>
+    props.transitionstate === 'exited' &&
+    `
+    opacity: 0;
+  `}
+`
+
+const Dialog = styled(DialogContent)`
+  ${color} 
+  ${width}
+  ${height}
+  max-width: calc(100vw - 32px);
+  position: relative;
+  margin-left: auto;
+  margin-right: auto;
+  box-shadow: ${props => props.theme.boxShadows[3]};
+  &:focus {
+    outline: none;
+  }
+  max-height: 100vh;
+  ${props =>
+    props.enableoverflow &&
+    `
+    max-height: none;
+  `}
+
+  transform: scale(0.5);
+  transition: transform .5s cubic-bezier(0.50, 0.00, 0.25, 1.00);
+
+  ${props =>
+    props.transitionstate === 'entering' &&
+    `
+    transform: scale(0.5);
+  `}
+
+  ${props =>
+    props.transitionstate === 'entered' &&
+    `
+    transform: scale(1);
+  `}
+
+  ${props =>
+    props.transitionstate === 'exiting' &&
+    `
+    transform: scale(0.5);
+  `}
+
+  ${props =>
+    props.transitionstate === 'exited' &&
+    `
+    transform: scale(0.5);
+  `}
+`
+
+const HeaderWrapper = styled.div`
+  flex-shrink: 0;
+`
+
+const FloatCloseButton = styled(CloseButton)`
+  position: absolute;
+  top: 16px;
+  right: 16px;
+  z-index: 2;
+
+  svg {
+    vertical-align: top;
+  }
+
+  &:focus {
+    outline: none;
+  }
+`
+
+const ContentWrapper = styled(Box)`
+  position: relative;
+  ${props => {
+    if (!props.enableoverflow) {
+      return `
+        overflow-y: scroll;
+        overflow-x: hidden;
+        height: 100%;
+        -webkit-overflow-scrolling: touch;
+      `
+    }
+  }}
+`
+
+const OverlayWrapper = styled.div`
+  display: table;
```

Comment:
Curious why you're using `display: table`.  Might this work as a Flex instead?

Reply:
no, the table here is a trick for always vertically center it, learned from airbnb modal